otter/test/convergence/test_service.py

```diff
@@ -152,34 +482,67 @@ def test_success(self):
         # yet. convergence should add them to the LBs.
         log = mock_log()
         gacd = self._get_gacd_func(self.group.uuid)
-        eff = execute_convergence(self.group, 2, self.lc, 0, log,
-                                  get_all_convergence_data=gacd)
-        mgs_eff = resolve_stubs(eff)
-        self._assert_active(mgs_eff, {})
-
-        eff = resolve_effect(mgs_eff, None)
-        # The steps are optimized
-        self.assertIsInstance(eff.intent, ParallelEffects)
-        self.assertEqual(len(eff.intent.effects), 1)
+        tscope_eff = execute_convergence(self.tenant_id, self.group_id, log,
+                                         get_all_convergence_data=gacd)
+        self.assertEqual(tscope_eff.intent.tenant_id, self.tenant_id)
+        self.assertEqual(tscope_eff.callbacks, [])
         expected_req = service_request(
             ServiceType.CLOUD_LOAD_BALANCERS,
             'POST',
             'loadbalancers/23/nodes',
-            data=mock.ANY,
+            data=transform_eq(
+                freeze,
+                pmap({
+                    'nodes': transform_eq(
+                        lambda nodes: set(freeze(nodes)),
+                        set([pmap({'weight': 1, 'type': 'PRIMARY',
+                                   'port': 80,
+                                   'condition': 'ENABLED',
+                                   'address': '10.0.0.2'}),
+                             pmap({'weight': 1, 'type': 'PRIMARY',
+                                   'port': 80,
+                                   'condition': 'ENABLED',
+                                   'address': '10.0.0.1'})]))})),
             success_pred=mock.ANY)
-        got_req = eff.intent.effects[0].intent
-        self.assertEqual(got_req, expected_req.intent)
-        # separate check for nodes; they are unique, but can be in any order
-        self.assertEqual(
-            set(freeze(got_req.data['nodes'])),
-            set([pmap({'weight': 1, 'type': 'PRIMARY', 'port': 80,
-                       'condition': 'ENABLED', 'address': '10.0.0.2'}),
-                 pmap({'weight': 1, 'type': 'PRIMARY', 'port': 80,
-                       'condition': 'ENABLED', 'address': '10.0.0.1'})]))
-        r = resolve_effect(eff, ['stuff'])
-        # The result of the parallel is returned directly
-        # TODO: This must change with issue #844.
-        self.assertEqual(r, ['stuff'])
+        expected_intents = self.expected_intents + [
+            (expected_req.intent, 'stuff')]
```

Comment:
Nitpick:  "stuff" -> "successful response" might make it more clear what is being stubbed out here.